idaes/core/base/extended_control_volume0d.py

```diff
@@ -0,0 +1,106 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES).
+#
+# Copyright (c) 2018-2024 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+#################################################################################
+"""
+0D Control Volume class with support for isothermal energy balance.
+"""
+
+__author__ = "Andrew Lee"
+
+# Import IDAES cores
+from idaes.core.base.control_volume0d import ControlVolume0DBlockData
+from idaes.core import declare_process_block_class
+from idaes.core.util.exceptions import ConfigurationError
+
+import idaes.logger as idaeslog
+
+_log = idaeslog.getLogger(__name__)
+
+
+@declare_process_block_class(
+    "ExtendedControlVolume0DBlock",
+    doc="""
+    ExtendedControlVolume0DBlock is an extension of the ControlVolume0D
+    block with support for isothermal conditions in place of a formal
+    energy balance.""",
+)
+class ExtendedControlVolume0DBlockData(ControlVolume0DBlockData):
+    """
+    Extended 0-Dimensional (Non-Discretized) ControlVolume Class
+
+    This class extends the existing ControlVolume0DBlockData class
+    with support for isothermal energy balances.
+    """
+
+    def add_isothermal_constraint(
+        self,
+        has_heat_of_reaction=False,
+        has_heat_transfer=False,
+        has_work_transfer=False,
+        has_enthalpy_transfer=False,
+        custom_term=None,
```

Comment:
Should we add Type Hints here?

Reply:
Might as well - if we don't do it now it probably won't get done.